sdk/include/reactos/drivers/bootvid/display.h

```diff
@@ -0,0 +1,12 @@
+/*
+ * PROJECT:     ReactOS video boot driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Provides display settings for BOOTVID, NTOSKRNL and HAL
+ * COPYRIGHT:   Copyright 2020 Dmitry Borisov (di.sean@protonmail.com)
+ */
+
+#pragma once
+
+/* For default VGA */
+#define SCREEN_WIDTH    640
```

Comment:
I'd move these to `inbv.h` as well to avoid including driver header in kernel and HAL (as I commented earlier).

Reply:
But if you do that we then completely loose the point of this PR?
These width & height are the only sizes currently imposed by bootvid that then need to be known by the other components using it (NTOSKRNL, HAL and BOOTVID itself), so I think they are fine here.

Reply:
Why? Just create a separate header file in `/inbv/` and include it where needed, just to make it clear - inbv controls the width and height, and bootvid driver imports header from it.

Reply:
Not sure I understand completely; the bootvid is the main limiter in this case and imposes the full screen width/height, that everything else using it gets, so it's more natural to have a global bootvid header that exposes these values, and code that uses bootvid also including this header.

Reply:
If you modify the header for different width/height and rebuild bootvid along with kernel and HAL, it would work in ReactOS case. But it won't work for Windows XP/2003, as its kernel will still use its own defined values, which makes me think the kernel (inbv component) is better place for such definitions. The bootvid driver has no way to tell the kernel its own preferred screen resolution, so we assume it's hardcoded, and if it's really hardcoded, it's done in the ntos:inbv.

Reply:
Just think of it like... the ntos:inbv is a "port", and bootvid is a "miniport". :)

Reply:
Not really. ntos:inbv is just a wrapper around bootvid, and the latter imposes its screen size.

Reply:
Alright then.

Reply:
hardcoded.h